src/game.cpp

```diff
@@ -11782,6 +11782,16 @@ bool game::plmove(int dx, int dy, int dz)
         dest_loc.z = u.posz() + dz;
     }
 
+    if( dest_loc == u.pos() ) {
+        // Well that sure was easy
+        return true;
+    }
+
+    if( dz == 0 && ramp_move( dest_loc ) ) {
```

Comment:
Ramp move invokes `plmove` recursively (hence that `dz == 0`, so it doesn't loop), this time it will trigger amigara with a new destination position.